src/gui/main.cpp

```diff
@@ -330,91 +329,110 @@ int main(int argc, char **argv)
-                }
-            }
-        });
+    FolderMan::instance()->setSyncEnabled(true);
 
-        if (options.show) {
-            ocApp->gui()->slotShowSettings();
-        }
-        if (!options.fileToOpen.isEmpty()) {
-            QTimer::singleShot(0, ocApp, [ocApp, fileToOpen = options.fileToOpen] { ocApp->openVirtualFile(fileToOpen); });
-        }
+    auto ocApp = new OCC::Application(platform.get(), options.debugMode, &app);
 
-        platform->startServices();
+    if (AccountManager::instance()->accounts().isEmpty()) {
+        // display the wizard if we don't have an account yet
+        QTimer::singleShot(0, ocApp->gui(), &ownCloudGui::runNewAccountWizard);
+    }
 
-#ifdef WITH_AUTO_UPDATER
-        // if handleStartup returns true, main()
-        // needs to terminate here, e.g. because
-        // the updater is triggered
-        Updater *updater = Updater::instance();
-        if (updater && updater->handleStartup()) {
-            return 1;
-        }
-#endif
+    QObject::connect(platform.get(), &Platform::requestAttention, ocApp->gui(), &ownCloudGui::slotShowSettings);
 
-    } else {
-        // if the application is already running, notify it.
-        qCInfo(lcMain) << "Already running, exiting...";
-        if (app.isSessionRestored()) {
-            // This call is mirrored with the one in Application::slotParseMessage
-            qCInfo(lcMain) << "Session was restored, don't notify app!";
-            return -1;
+    QObject::connect(&singleApplication, &KDSingleApplication::messageReceived, ocApp, [&](const QByteArray &message) {
+        const QString msg = QString::fromUtf8(message);
+        qCInfo(lcMain) << Q_FUNC_INFO << msg;
+        if (msg.startsWith(msgParseOptionsC())) {
+            const QStringList optionsStrings = msg.mid(msgParseOptionsC().size()).split(QLatin1Char('|'));
+            CommandLineOptions options = parseOptions(optionsStrings);
+            if (options.show) {
+                ocApp->gui()->slotShowSettings();
+            }
+            if (options.quitInstance) {
+                qApp->quit();
+            }
+            if (!options.fileToOpen.isEmpty()) {
+                QTimer::singleShot(0, ocApp, [ocApp, fileToOpen = options.fileToOpen] { ocApp->openVirtualFile(fileToOpen); });
+            }
         }
+    });
 
-        QStringList args = app.arguments();
-        if (args.size() > 1) {
-            QString msg = args.join(QLatin1String("|"));
-            if (!singleApplication.sendMessage((msgParseOptionsC() + msg).toUtf8()))
-                return -1;
-        }
-        return 0;
+    if (options.show) {
+        ocApp->gui()->slotShowSettings();
     }
+    if (!options.fileToOpen.isEmpty()) {
+        QTimer::singleShot(0, ocApp, [ocApp, fileToOpen = options.fileToOpen] { ocApp->openVirtualFile(fileToOpen); });
+    }
+
+    platform->startServices();
+
+#ifdef WITH_AUTO_UPDATER
+    // if handleStartup returns true, main()
```

Comment:
Hmm handling the updater AFTER we loaded everything/displayed a wizard looks wrong